### DISCUSSION TITLE
fake pages reset when moving back after leaving

### DISCUSSION THREAD

**Discussion by ninaassmann** [neuefische/questions#710] [Web] [answered]
I have an array for dog breeds which I split in smaller pieces to have a better performance.
For that I am using array.slice(start, end).map...
I have the slice options in localStorage

I also have 3 filters to filter the breeds. These filter update the array, which will be mapped to display the breeds.
This is also in localStorage. I managed to resetthe slice options, when a filter is set, so the user directly gets to "page" one and see the first filtered breed on top.

When the user clicks on one breed at another page than the first and then go back to the overwiev the filter are still set and the right breeds are visible, but sadly it jumps back to page one.

I made a pull request for that
https://github.com/ninaassmann/nf-capstone-project/pull/27

the important code snippets are this
the state for the filter:
```
  const [filter, setFilter] = useLocalStorageState("breedFilter", {
    defaultValue: initialFilter,
  });
```
the state for the sliceOptions
```
  const [sliceOptions, setSliceOptions] = useLocalStorageState("sliceOptions", {
    defaultValue: initialSliceOptions,
  });
```
here is how I set the array
```
  const breedsToShow = dogBreedFilter(filter, dogBreeds && dogBreeds);
  const breedsToShowCount = breedsToShow?.length;
```
I use the effect hook to reset the sliceOptions when the filter changed, but it seems, when I switch pages and go back to the overview, this also triggers
```
  useEffect(() => {
    setSliceOptions(initialSliceOptions);
  }, [setSliceOptions, filter]);
```

**Comment by ninaassmann** [comment]
when I comment the useEffect it's working, but then I have an issue with the filter
when I use the filter and I want the pages to reset

**Reply by milodevhh** [reply]
Just look at it again yourself ... you are super

**Comment by ninaassmann** [comment, marked as answer]
resolved it myself
made an onChange function for my filters and reset the sliceOptions in there

```
  function handleOnChange(event, key) {
    const newFilter = {
      ...filter,
    };
    newFilter[key] = event.target.value;
    setFilter(newFilter);
    setSliceOptions(initialSliceOptions);
  }
```

**Reply by MerleMak** [reply]
Amazing!! Good job 🚀
